### PR TITLE
sliding sync: Trigger a `RoomList` update when a room is updated b/o extensions

### PR DESCRIPTION
**Pull request by bnjbvr**
Now with a nice test :partying_face: 

Some sliding sync responses may include extension data that will cause an update to the room. For instance, whenever we receive a read receipt for that room, the content of the update lives in the extensions subpart of the response, and it may not involve an entry in the `rooms` or `lists` response fields. This could cause missed updates for rooms, since notifiers wouldn't be notified about those updates, until another response came that mentioned the room in one of the `rooms` or `lists` fields.

This fixes that, by making sure that we take into account room implicitly updated, as observed by the rooms returned in the base client's sync response. The test exhibits the issue (it doesn't pass on main).

### CODE REVIEW

File: crates/matrix-sdk/src/sliding_sync/mod.rs [hunk 1]
```diff
@@ -412,18 +421,28 @@ impl SlidingSync {
                 let mut updated_lists = Vec::with_capacity(sliding_sync_response.lists.len());
                 let mut lists = self.inner.lists.write().await;
 
-                for (name, updates) in sliding_sync_response.lists {
-                    let Some(list) = lists.get_mut(&name) else {
-                        error!("Response for list `{name}` - unknown to us; skipping");
+                // Iterate on known lists, not on lists in the response. Rooms may have been
+                // updated that were not involved in any list update.
```
Comment (Hywan) [inline crates/matrix-sdk/src/sliding_sync/mod.rs:425]: :+1:

File: crates/matrix-sdk/src/sliding_sync/list/mod.rs [hunk 2]
```diff
@@ -260,6 +260,12 @@ impl SlidingSyncList {
     pub fn sync_mode(&self) -> SlidingSyncMode {
         self.inner.sync_mode.read().unwrap().clone()
     }
+
+    /// Sets some rooms in the filled state.
+    pub fn set_filled_rooms(&self, rooms: Vec<OwnedRoomId>) {
```
Comment (Hywan) [inline crates/matrix-sdk/src/sliding_sync/list/mod.rs:271]: Do we want this to be part of the public API? I don't believe so. Can we switch to `pub(crate)`?

**Review by Hywan — CHANGES_REQUESTED** [review]
Please address my feedback, otherwise it looks good! Thanks!

File: crates/matrix-sdk/src/sliding_sync/list/mod.rs (hunk 2 above)
Comment (bnjbvr) [reply crates/matrix-sdk/src/sliding_sync/list/mod.rs:271]: This is defined in a `#[cfg(testing)]` block, so I was supposing that's fine?

File: crates/matrix-sdk/src/sliding_sync/mod.rs (hunk 1 above)
Comment (bnjbvr) [reply crates/matrix-sdk/src/sliding_sync/mod.rs:425]: :+1:

Comment (Hywan) [reply crates/matrix-sdk/src/sliding_sync/mod.rs:425]: :+1:

File: crates/matrix-sdk/src/sliding_sync/list/mod.rs (hunk 2 above)
Comment (Hywan) [reply crates/matrix-sdk/src/sliding_sync/list/mod.rs:271]: Yeah! Of course! Sorry for the noise.

**Review by Hywan — APPROVED** [review]

**Pull request merged**